### PR TITLE
feat: Add forms

### PR DESCRIPTION
**Pull request by piero-vic**
# feat: Add forms

- Add forms for creating posts and comments.
- Add button to like a post.
- Add website header.

NOTE: This project uses TailwindCSS. Please make sure to run `./bin/dev`. This is needed for generating the CSS.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -7,7 +7,23 @@
...
+  <%= form_with model: @comment, url: user_post_comments_path(@post.author.id, @post.id) do |f| %>
+    <div class="border p-3 flex flex-col gap-2">
+      <h2 class="text-xl">Create a new comment</h2>
+      <div>
+        <%= f.label :text, "Comment", class: "sr-only" %>
+        <%= f.text_area :text, required: true, class: "w-full p-2 border" %>
+      </div>
+      <%= f.submit "Post Comment", class: "self-end py-2 px-3 border rounded-xl" %>
+    </div>
+  <% end %>
```
Comment (andersonlebon) [inline app/views/posts/show.html.erb:28]: - [ ] Kindly this form should not be here rather it should be in the the `comments/new.html.erb` as the requirements say.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      redirect_to user_post_path(user_id, post_id), alert: 'Something went wrong'
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+end
```
Comment (andersonlebon) [inline app/controllers/comments_controller.rb:21]: - [ ] kindly as the requirements said you need to create an action(new) for rendering the comment form. This means you need also to add a view(comment#new) for rendering the form for creating a new comment.

```ruby
  def new
    @comment = Comment.new
  end
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def create
+    @current_user = current_user
+    user_id = params[:user_id]
+    post_id = params[:post_id]
+
+    comment = Comment.new(comment_params.merge(author_id: @current_user.id, post_id:))
```
Comment (andersonlebon) [inline app/controllers/comments_controller.rb:7]: image

- [ ] Kindly note that this is throwing out a syntax error, kindly fix it.

Hint :point_down: 

```rb
 comment = Comment.new(comment_params.merge(author_id: @current_user.id, post_id: post_id))
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @piero-vic 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @piero-vic, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**